### DISCUSSION TITLE
Вывод значений нескольких полей через "сепаратор"

### DISCUSSION THREAD

**Discussion by mihail-cornev** [totumonline/totum-ru-issues-and-discussions#288] [Как это сделать?] [answered]
Задача:

Собрать в одну строку значения из разных полей через сепаратор, например запятая.
Например есть несколько полей:
`#city #street #name #surname #nickname`

Бывает так, что не все поля заполнены и у меня выходит только дублирование запятых, типа:
`City, , Vasily, , Vasea`
Или иногда вообще выходит
`, , , , Vasea`

Думал через strSplit, но с этой функцией вообще не разобрался.

Спасибо.

**Comment by erpic-ru** [comment, marked as answer]
Если каждый параметр обернуть в переменную c запятой, то будет норм

Например:

```
= : str`$city + $street + ...` 

city: if(condition: #city!=""; then: $city_str)

city_str: str`#city + ", "`
```

Ну и второй вариант: 
 
```
=: listJoin(list: $1; str:", " )

    1:listcreate (item: #city; item: #street....)

```

UPD: Уточнения внизу 👇

**Reply by mihail-cornev** [reply]
какое простое и хорошее решение ) спасибо, ща буду тестить

**Reply by ttm-alexeypolunin** [reply]
Ток надо фильтрануть на пустые значения:

```
= : listJoin(list: $filtered; str: ", ")

filtered: listFilter(list: $list; key: "value" != "")
	list: listCreate(item: #books; item: #first_attempt; item: #second_attempt)
```

**Reply by ttm-alexeypolunin** [reply]
А еще, если использовать str`` то можно if не множить, а вызывать одну строку с переменной:

```
= : str`#books + $if{var: "name" = "books"} + #first_attempt + $if{var: "name" = "first_attempt"} + #second_attempt`

if: if(condition: #$field != ""; then: ", "; else: "")

	field: $#name
```

Те мы передаем в переменную `field` имя поля и потом в строке if берем значение этого поля.

Но обычно это используется для более сложных случаев, чем простановка запятой. И переменных кстати можно в `{...}` передать несколько через `;`.

**Reply by mihail-cornev** [reply]
> = : str`#books + $if{var: "field" = "books"} + #first_attempt + $if{var: "field" = "first_attempt"} + #second_attempt`
> 
> if: if(condition: #$field != ""; then: ", "; else: "")
> 
> 	field: $#field

выдаёт ошибку. не пойму где поправить. #$field $#field ?

**Reply by ttm-alexeypolunin** [reply]
По разному назвал — может так понятнее будет

**Reply by mihail-cornev** [reply]
> По разному назвал — может так понятнее будет

не, я не понимаю... логически я понимаю, что должно быть, но код... 
почему #$field тут решетка спереди
а тут $#name - а тут после доллара

**Reply by ttm-alexeypolunin** [reply]
+ `#$field` — взять значение из поля `#`, нейм которого будет определен в строке кода `$field`

+ `$#name` — получить значение переменной внутри кода обозначенной как `name`.

Когда мы вызываем `$if{var: "name" = "books"}` мы создаем переменную `name` и назначаем ей значение `"books"`.

И так как мы 2 раза вызываем строку `$if` с разными значениями переменной — она вернет разные результаты.

**Reply by mihail-cornev** [reply]
> * `#$field` — взять значение из поля `#`, нейм которого будет определен в строке кода `$field`
> * `$#name` — получить значение переменной внутри кода обозначенной как `name`.
> 
> Когда мы вызываем `$if{var: "name" = "books"}` мы создаем переменную `name` и назначаем ей значение `"books"`.
> 
> И так как мы 2 раза вызываем строку `$if` с разными значениями переменной — она вернет разные результаты.

Спасибо за помощь и за терпение.

**Comment by erpic-ru** [comment]
У меня есть ещё способ через listreplace :), но кажется, что достаточно уже

**Reply by ttm-alexeypolunin** [reply]
Реплейсить список и добавлять в значение. Но это если разное надо добавить в зависимости от значения в шаге.
